pkg/inventory/inventory.go

```diff
@@ -73,16 +72,7 @@ func IsInventoryObject(obj *unstructured.Unstructured) bool {
 // for the passed inventory object. Returns error if the passed object is nil or
 // is not a inventory object.
 func retrieveInventoryLabel(obj *unstructured.Unstructured) (string, error) {
-	if obj == nil {
```

Comment:
Cannot be `nil` here.

Reply:
Up to you if you want to remove these checks. There is disagreement here, but many consider defensively validating all parameters good practice. These validation checks make the contract between the caller and callee more concrete.

Reply:
I think in Go the default assumption is that parameters are not `nil` (for pointers). I don't think you can often find such checks in code, at least not in the Kubernetes codebase. Also, this is a private function and arguably the check is not needed at all. This check is like an assertion rather than input validation. But the code will crash immediately anyway if a `nil` is passed, so no need for an extra assertion. That was my reasoning :)